src/wallet/asyncrpcoperation_mergetoaddress.cpp

```diff
@@ -0,0 +1,943 @@
+                           strprintf("Insufficient funds, have %s and miners fee is %s",
+                                     FormatMoney(targetAmount), FormatMoney(minersFee)));
+    }
+
+    CAmount sendAmount = targetAmount - minersFee;
+
+    // update the transaction with the UTXO inputs and output (if any)
+    CMutableTransaction rawTx(tx_);
+    for (MergeToAddressInputUTXO& t : utxoInputs_) {
+        CTxIn in(std::get<0>(t));
+        rawTx.vin.push_back(in);
+    }
+    if (isToTaddr_) {
+        CScript scriptPubKey = GetScriptForDestination(toTaddr_.Get());
+        CTxOut out(sendAmount, scriptPubKey);
+        rawTx.addOut(out);
+    }
+    tx_ = CTransaction(rawTx);
+
+    LogPrint(isPureTaddrOnlyTx ? "zrpc" : "zrpcunsafe", "%s: spending %s to send %s with fee %s\n",
+             getId(), FormatMoney(targetAmount), FormatMoney(sendAmount), FormatMoney(minersFee));
+    LogPrint("zrpc", "%s: transparent input: %s\n", getId(), FormatMoney(t_inputs_total));
+    LogPrint("zrpcunsafe", "%s: private input: %s\n", getId(), FormatMoney(z_inputs_total));
+    if (isToTaddr_) {
+        LogPrint("zrpc", "%s: transparent output: %s\n", getId(), FormatMoney(sendAmount));
+    } else {
+        LogPrint("zrpcunsafe", "%s: private output: %s\n", getId(), FormatMoney(sendAmount));
+    }
+    LogPrint("zrpc", "%s: fee: %s\n", getId(), FormatMoney(minersFee));
+
+    /**
+     * SCENARIO #1
+     *
+     * taddrs -> taddr
+     *
+     * There are no zaddrs or joinsplits involved.
+     */
+    if (isPureTaddrOnlyTx) {
+        UniValue obj(UniValue::VOBJ);
+        obj.pushKV("rawtxn", EncodeHexTx(tx_));
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #1
+     */
+
+
+    // Prepare raw transaction to handle JoinSplits
+    CMutableTransaction mtx(tx_);
+    crypto_sign_keypair(joinSplitPubKey_.begin(), joinSplitPrivKey_);
+    mtx.joinSplitPubKey = joinSplitPubKey_;
+    tx_ = CTransaction(mtx);
+    std::string hexMemo = std::get<1>(recipient_);
+
+
+    /**
+     * SCENARIO #2
+     *
+     * taddrs -> zaddr
+     *
+     * We only need a single JoinSplit.
+     */
+    if (noteInputs_.empty() && isToZaddr_) {
+        // Create JoinSplit to target z-addr.
+        MergeToAddressJSInfo info;
+        info.vpub_old = sendAmount;
+        info.vpub_new = 0;
+
+        JSOutput jso = JSOutput(toPaymentAddress_, sendAmount);
+        if (hexMemo.size() > 0) {
+            jso.memo = get_memo_from_hex_string(hexMemo);
+        }
+        info.vjsout.push_back(jso);
+
+        UniValue obj(UniValue::VOBJ);
+        obj = perform_joinsplit(info);
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #2
+     */
+
+
+    // Copy zinputs to more flexible containers
+    std::deque<MergeToAddressInputNote> zInputsDeque;
+    for (auto o : noteInputs_) {
+        zInputsDeque.push_back(o);
+    }
```

Comment:
I'd suggest the following form, which is more compact and saves some mem copies:

```suggestion
    std::deque<MergeToAddressInputNote> zInputsDeque(noteInputs_.begin(), noteInputs_.end());
```